examples/hosted-delivery-function-templates/javascript/build-delivery-functions.js

```diff
@@ -0,0 +1,81 @@
+const esbuild = require('esbuild');
+const { join, parse, resolve } = require('path');
+const yargs = require('yargs/yargs');
+const { hideBin } = require('yargs/helpers');
+
+const manifest = require('./contentful-app-manifest.json');
+
+const argv = yargs(hideBin(process.argv)).argv;
+
+const validateActions = () => {
+  const requiredProperties = ['id', 'path', 'entryFile'];
+  const uniqueValues = new Set();
+
+  manifest.deliveryFunctions.forEach((deliveryFunction) => {
+    requiredProperties.forEach((property) => {
+      if (!deliveryFunction.hasOwnProperty(property)) {
+        throw new Error(
+          `Delivery function with name: '${deliveryFunction.name}' is missing the '${property}' property`
+        );
+      }
+    });
+
+    const { id, path, entryFile } = deliveryFunction;
+
+    if (uniqueValues.has(id)) {
+      throw new Error(`Duplicate action id: '${id}'`);
+    }
+    if (uniqueValues.has(path)) {
```

Comment:
curious, couldn't there be a scenario where two functions share the same file? We can loosen up this part in the future